solidity/test/AcreMultiAssetVault.test.ts

```diff
@@ -0,0 +1,949 @@
+                initialMezoDepositCount + deposit1.depositId,
+              )
+          })
+
+          it("should delete deposit data", async () => {
+            const depositData = await multiAssetVault.getDeposit(
+              await deposit1.depositOwner.getAddress(),
+              await deposit1.asset.getAddress(),
+              deposit1.depositId,
+            )
+
+            expect(depositData.balance).to.be.equal(0)
+            expect(depositData.mezoDepositId).to.be.equal(0)
+          })
+
+          it("should transfer the asset to the receiver", async () => {
+            await expect(tx).to.changeTokenBalances(
+              asset1,
+              [mezoPortal, multiAssetVault, receiver],
+              [-deposit1.depositAmount, 0, deposit1.depositAmount],
+            )
+          })
+
+          context("when tries to withdraw the same deposit again", () => {
+            it("should revert", async () => {
+              await expect(
+                multiAssetVault
+                  .connect(deposit1.depositOwner)
+                  .withdraw(
+                    await deposit1.asset.getAddress(),
+                    deposit1.depositId,
+                    await receiver.getAddress(),
+                  ),
+              ).to.be.revertedWithCustomError(
+                multiAssetVault,
+                "DepositNotFound",
+              )
+            })
+          })
+        })
+      })
+    })
+
+    context("when multiple deposits are withdrawn", () => {
+      beforeAfterSnapshotWrapper()
+
+      let deposit1: Deposit
+      let deposit2: Deposit
+      let deposit3: Deposit
+
+      before(async () => {
+        deposit1 = {
+          caller: depositor1,
+          asset: asset1,
+          depositAmount: to1e18(10),
+          depositOwner: depositor2,
+          depositId: 1n,
+        }
+        await createDepositFor(deposit1)
+
+        deposit2 = {
+          caller: depositor2,
+          asset: asset2,
+          depositAmount: to1e18(11),
+          depositOwner: depositor1,
+          depositId: 2n,
+        }
+        await createDepositFor(deposit2)
+
+        deposit3 = {
+          caller: depositor1,
+          asset: asset1,
+          depositAmount: to1e18(12),
+          depositOwner: depositor1,
+          depositId: 3n,
+        }
+        await createDepositFor(deposit3)
+      })
+
+      it("should withdraw correct deposit 2", async () => {
+        await testWithdrawDeposit(deposit2)
+      })
+
+      it("should withdraw correct deposit 1", async () => {
+        await testWithdrawDeposit(deposit1)
+      })
+
+      it("should withdraw correct deposit 3", async () => {
+        await testWithdrawDeposit(deposit3)
+      })
+
+      async function testWithdrawDeposit({
+        asset,
+        depositAmount: expectedWithdrawAmount,
+        depositOwner,
+        depositId,
+      }: Deposit & { depositId: bigint; receiver: HardhatEthersSigner }) {
```

Comment:
I am confused, where do we pass the value for the `receiver`?

Reply:
This entire construction is a bit twisted. I wonder if we can't just add `depositId` field to the `Deposit` struct.